### PR TITLE
feat: destroy ads if return now number of dropped items

### PR DESCRIPTION
**Pull request by florent37**
Returns the number of items being deleted by the call to `destroyAdsIf`

### CODE REVIEW

File: ads-api/src/main/java/io/voodoo/apps/ads/api/AdClient.kt [hunk 1]
```diff
@@ -284,17 +287,22 @@ abstract class BaseAdClient<ActualType : PublicType, PublicType : Ad>(
         adClickListeners.remove(listener)
     }
 
-    override fun destroyAdsIf(predicate: (Ad) -> Boolean) {
+    /**
+     * @return number of elements being destroyed
+     */
```
Comment (badoualy) [inline ads-api/src/main/java/io/voodoo/apps/ads/api/AdClient.kt:292]: no need to copy doc here

File: ads-api/src/main/java/io/voodoo/apps/ads/api/AdClient.kt [hunk 2]
```diff
@@ -284,17 +287,22 @@ abstract class BaseAdClient<ActualType : PublicType, PublicType : Ad>(
...
+    override fun destroyAdsIf(predicate: (Ad) -> Boolean) : Int {
         synchronized(loadedAds) {
             val renderedAds = loadedAds.filter(predicate)
```
Comment (florent37) [inline ads-api/src/main/java/io/voodoo/apps/ads/api/AdClient.kt:295]: hum maybe rename the "renderedAds" to "adsToDestroy"

File: ads-api/src/main/java/io/voodoo/apps/ads/api/AdClient.kt [hunk 3]
```diff
@@ -77,7 +77,10 @@ interface AdClient<T : Ad> : Closeable, AdListenerHolder {
     fun addOnAvailableAdCountChangedListener(listener: OnAvailableAdCountChangedListener)
     fun removeOnAvailableAdCountChangedListener(listener: OnAvailableAdCountChangedListener)
 
-    fun destroyAdsIf(predicate: (Ad) -> Boolean)
+    /**
```
Comment (badoualy) [inline ads-api/src/main/java/io/voodoo/apps/ads/api/AdClient.kt:80]:
```kotlin
    /**
     * @return number of elements being destroyed
     *
     * Note: if an ad is locked (being used somewhere after a [getAvailableAd] call), 
     * it won't get destroyed until being released via a call to [releaseAd]
     */
```

File: ads-api/src/main/java/io/voodoo/apps/ads/api/AdClient.kt [hunk 4]
```diff
@@ -284,17 +287,22 @@ abstract class BaseAdClient<ActualType : PublicType, PublicType : Ad>(
...
         synchronized(loadedAds) {
-            val renderedAds = loadedAds.filter(predicate)
+            val adsToDestroy = loadedAds.filter(predicate)
 
-            val renderedAdsNotLocked = loadedAds.filter { it.isLocked().not() }
+            val adsToDestroyNotLocked = loadedAds.filter { it.isLocked().not() }
             pendingAdsToRelease.addAll(loadedAds.filter { it.isLocked() }.map { it.id })
 
-            if (renderedAdsNotLocked.isEmpty()) return
+            if (adsToDestroyNotLocked.isEmpty()) return adsToDestroy.size
```
Comment (florent37) [inline ads-api/src/main/java/io/voodoo/apps/ads/api/AdClient.kt:300]: whatever they are locked or not, the size should be `adsToDestroy.size` not `adsToDestroyNotLocked.size`

Comment (badoualy) [reply ads-api/src/main/java/io/voodoo/apps/ads/api/AdClient.kt:300]: yup agree

File: ads-api/src/main/java/io/voodoo/apps/ads/api/AdClient.kt (hunk 2 above)
Comment (badoualy) [reply ads-api/src/main/java/io/voodoo/apps/ads/api/AdClient.kt]: inded

**Review by badoualy — APPROVED** [review]

**Pull request merged**